include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -230,41 +248,39 @@ std::string get_cigar_string(ref_seq_type && ref_seq,
+                                                  bool const extended_cigar = false)
 //!\cond
     requires std::tuple_size_v<remove_cvref_t<alignment_type>> == 2
 //!\endcond
-std::string get_cigar_string(alignment_type && alignment,
-                             uint32_t const query_start_pos = 0,
-                             uint32_t const query_end_pos = 0,
-                             bool const extended_cigar = false)
 {
-    return get_cigar_string(get<0>(alignment), get<1>(alignment),
-                            query_start_pos, query_end_pos, extended_cigar);
+    return get_cigar_string(get_cigar_vector(alignment, query_start_pos, query_end_pos, extended_cigar));
 }
-/*!\brief Creates a CIGAR string (SAM format) given an alignment represented by two aligned sequences.
+
+/*!\brief Transforms an alignment represented by two aligned sequences into the
+ *        corresponding CIGAR string.
  * \ingroup alignment_file
  *
- * \tparam alignment_type  Must model the seqan3::tuple_like and must
- *                         have std::tuple_size 2. Each tuple element must model
- *                         std::forward_range and its value_type must be comparable
- *                         to seqan3::gap.
- * \param  alignment       The alignment, represented by a pair of aligned sequences,
- *                         to be transformed into CIGAR_vector based on the
- *                         second (query) sequence.
+ * \tparam ref_seq_type    Must model std::ranges::forward_range. The value_type must
+ *                         be equality comparable to seqan3::gap.
+ * \tparam query_seq_type  Must model std::ranges::forward_range. The value_type must
+ *                         be equality comparable to seqan3::gap.
+ * \param  ref_seq         The reference sequence to compare against the query sequence.
+ * \param  query_seq       The query sequence to build the CIGAR string for.
  * \param  query_start_pos The start position of the alignment in the query
  *                         sequence indicating soft-clipping.
  * \param  query_end_pos   The end position of the alignment in the query
  *                         sequence indicating soft-clipping.
  * \param  extended_cigar  Whether to print the extended cigar alphabet or not. See CIGAR operation.
- * \returns An std::vector<std::pair<char, size_t>> representing the alignment as a pairs of cigar_op and cigar count.
+ * \returns An std::string representing the alignment as a CIGAR string.
  *
  * \details
  *
- * \attention Note that CIGAR elements (respectively by their
- *            CIGAR operation) are always related to on one of the two sequences
- *            in a pairwise alignment. In this case, the resulting cigar_vector
- *            is based on sequence at the second position of the \p alignment pair,
- *            namely the query sequence.
+ * \attention Note that CIGAR elements (respectively by their CIGAR operation) are always related to one of the two
```

Comment:
that would be also a candidate not to miss out if you agree on changing the description as proposed.